### PR TITLE
updated the shape of state and actions

### LINKED ISSUE #140: Update create jobpost
We need to save the response after creating a job post

### PR DESCRIPTION
**Pull request by dilippoudel**
Fix #140

### CODE REVIEW

File: src/redux/actions/jobpost.ts [hunk 1]
```diff
@@ -15,9 +15,10 @@ export const creatingJobPostRequest = (
     payload: jobForm,
   }
 }
-export const registerJobPostSuccess = (): JobSuccessActionType => {
+export const registerJobPostSuccess = (data: any): JobSuccessActionType => {
```
Comment (Wyfy0107) [inline src/redux/actions/jobpost.ts]: I think `jobPost` would be more descriptive than `data`, and try to give a type also. You can `console.log` to know its type.

File: src/redux/reducers/employer.ts [hunk 2]
```diff
@@ -23,6 +23,8 @@ const initialState = {
     seniority: '',
     skills: [],
   },
+  jobLists: [], // this is store to list of all job poste created by the logged in employer
```
Comment (Wyfy0107) [inline src/redux/reducers/employer.ts:26]: We already have a `resource` state inside the store. It's better to save it in there:
`resource: { jobLists: [] }`

File: src/redux/saga/employer.ts [hunk 3]
```diff
@@ -38,7 +38,7 @@ function* creatingJobPostSaga() {
     const job = yield select(jobPostFormData)
     const res = yield axios.post('/employer/jobs', job)
     console.log(res)
```
Comment (Wyfy0107) [inline src/redux/saga/employer.ts]: Cleanup

**Review by Wyfy0107 — CHANGES_REQUESTED** [review]

File: src/redux/reducers/employer.ts (hunk 2 above)
Comment (dilippoudel) [reply src/redux/reducers/employer.ts:26]: resource have only skill sets, not for joblists

Comment (Wyfy0107) [reply src/redux/reducers/employer.ts:26]: The idea of `resource` is to store all the response data that will be used to display. So both `skillSets` and `jobLists` can be inside `resource`. Later we can put more stuff inside `resources` as well

Comment (dilippoudel) [reply src/redux/reducers/employer.ts:26]: okay i see now there is a different reducer for resources already :-D

File: src/redux/actions/jobpost.ts [hunk 4]
```diff
@@ -15,7 +16,9 @@ export const creatingJobPostRequest = (
     payload: jobForm,
   }
 }
-export const registerJobPostSuccess = (jobPost: any): JobSuccessActionType => {
+export const registerJobPostSuccess = (
+  jobPost: dataJobPostType
```
Comment (Wyfy0107) [inline src/redux/actions/jobpost.ts:20]: `DataJobPostType`
Try to use Pascal case for type names

**Review by Wyfy0107 — CHANGES_REQUESTED** [review]
Can you bring `jobList` state inside `resource`?

**Pull request closed without merging**